src/dispatch/__tests__/DataPlaneClient.test.ts

```diff
@@ -132,4 +132,107 @@ describe('DataPlaneClient tests', () => {
             'd37eb756444ebf6f785233714d6d942f2b20f69292fb09533f6b69556eb0ff2b'
         );
     });
+
+    test('when the endpoint contains a path then the fetch reqest url contains the path prefix', async () => {
+        // Init
+        const endpoint = new URL(
+            `${Utils.AWS_RUM_ENDPOINT.toString()}${'prod'}`
+        );
+        const client: DataPlaneClient = new DataPlaneClient({
+            fetchRequestHandler: new FetchHttpHandler(),
+            beaconRequestHandler: new BeaconHttpHandler(),
+            endpoint,
+            region: Utils.AWS_RUM_REGION,
+            credentials: Utils.createAwsCredentials()
+        });
+
+        // Run
+        await client.sendFetch(Utils.PUT_RUM_EVENTS_REQUEST);
+
+        // Assert
+        const signedRequest: HttpRequest = (fetchHandler.mock
+            .calls[0] as any)[0];
+        expect(signedRequest.hostname).toEqual(Utils.AWS_RUM_ENDPOINT.hostname);
```

Comment:
question: Could we add examples for specific parameters, such as hostname, host, path, etc as comments?

Reply:
What value would this add? The test is checking one property of the code.

Reply:
Perhaps the better place might be in the dataplane client code... The idea was to show how we parse a URL into `hostname`, `host`, `path`, to help contributors understand how we create HTTP requests - just a thought

Reply:
Added a line to the user docs about the feature.

URL parsing is standardized/defined by W3C -- should be clear.